aioredlock/algorithm.py

```diff
@@ -177,7 +188,12 @@ def log(self):
 
     async def destroy(self):
         """
-        Clear all the redis connections
+        cancel all watchdogs and Clear all the redis connections
         """
         self.log.debug('Destroying %s', repr(self))
+
+        for resource in self.watchdogs:
+            self.watchdogs[resource].cancel()
+        self.watchdogs = {}
```

Comment:
All spawned tasks should be awaited. Consider to await all tasks, suppress asyncio.CancelError, suppress and log any other exceptions.
```python
for fut in self.watchdogs.values():
    try:
        await fut
    except asyncio.CancelledError:
        pass
    except Exception:
        self.log.exception("Unhanded error in lock watchdog.")
```